### PR TITLE
[graph] Bug fix for duplicate output connections

### LINKED ISSUE #1140: Graph nodes have duplicate output layers
Graph nodes have duplicate output layers.
This does not cause any issues for now, but its best to solve this soon.

### PR DESCRIPTION
**Pull request by kparichay**
Added bugfix for duplicate output connections

Resolves #1140

**Self evaluation:**
1. Build test: [x]Passed [ ]Failed [ ]Skipped
2. Run test: [x]Passed [ ]Failed [ ]Skipped

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

File: nntrainer/graph/network_graph.cpp [hunk 1]
```diff
@@ -468,11 +468,16 @@ void NetworkGraph::setOutputLayers() {
...
       for (unsigned int j = 0; j < layer_i->input_layers.size(); ++j) {
         if (istrequal(layer_i->input_layers[j], layer_idx->getName())) {
+          bool already_exist = false;
           for (unsigned int k = 0; k < layer_idx->output_layers.size(); ++k) {
-            if (!istrequal(layer_idx->output_layers[k], layer_i->getName()))
-              continue;
+            if (!istrequal(layer_idx->output_layers[k], layer_i->getName())) {
+              already_exist = true;
+              break;
+            }
```
Comment (zhoonit) [inline nntrainer/graph/network_graph.cpp:476]: slightly off the topic though, I wonder what this loop is trying to to. It look essentially noop to me.

Comment (kparichay) [reply nntrainer/graph/network_graph.cpp:476]: it is trying to avoid duplicates in `layer_idx->output_layers`.

Comment (zhoonit) [reply nntrainer/graph/network_graph.cpp:476]: Ah, I see, nice catch 👍

**Review by zhoonit — APPROVED** [review]

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Pull request merged**